src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,122 @@
 package mate.academy.lib;
 
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class Injector {
+    public static final String PATH_DELIMITER = "/";
+    public static final String CLASSNAME_DELIMITER = ".";
     private static final Injector injector = new Injector();
+    private static final String SERVICE_IMPL_PACKAGE = "mate/academy/service/impl";
+    private static final String SERVICE_PACKAGE = "mate/academy/service";
+    private static final String CLASS_EXTENSION = ".class";
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> implementations;
+
+    private Injector() {
+        List<Class<?>> interfaces = findInterfaces();
+        implementations = findImplementations(interfaces);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implementationClazz = implementations.get(interfaceClazz);
+        boolean isPresent = instances.containsKey(implementationClazz);
+        if (isPresent) {
+            return instances.get(implementationClazz);
+        }
+        if (implementationClazz != null) {
+            try {
+                Object o = implementationClazz.getDeclaredConstructor().newInstance();
+                Field[] fields = o.getClass().getDeclaredFields();
+                for (Field field : fields) {
+                    Annotation[] annotations = field.getAnnotations();
+                    if (isAnnotationPresent(annotations, Inject.class)) {
+                        field.setAccessible(true);
+                        field.set(o, getInstance(field.getType()));
+                        field.setAccessible(false);
+                    }
+                }
+                instances.put(implementationClazz, o);
+                return o;
+
+            } catch (ReflectiveOperationException e) {
+                throw new RuntimeException(
+                        String.format("Failed to create instance of %s", interfaceClazz));
```

Comment:
The error message in this `RuntimeException` is too generic. Consider including more specific information about the failure, such as the class that failed to instantiate or the specific exception message.